flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/table/FileSystemTableSource.java

```diff
@@ -225,16 +222,27 @@ public ScanRuntimeProvider getScanRuntimeProvider(ScanContext scanContext) {
             List<ReadableFileInfo> metadata,
             List<String> partitionKeys) {
         if (!metadata.isEmpty() || !partitionKeys.isEmpty()) {
+            final List<String> producedFieldNames = DataType.getFieldNames(producedDataType);
+            final Map<String, FileInfoAccessor> metadataColumns =
+                    IntStream.range(0, metadata.size())
+                            .mapToObj(
+                                    i -> {
+                                        // Access metadata columns from behind because the
```

Comment:
from behind -> `backwards`, or `in reverse` ?

Reply:
I looked it up and chose `from the back`